animation-proposal.md

```diff
@@ -0,0 +1,338 @@
+	 */
+	public name: string;
+
+	/**
+	 * Starts playing the animation from the beginning.
+	 * @returns A Promise that resolves when the animation completes or is stopped.
+	 */
+	public start(): Promise<void>;
+
+	/**
+	 * Stops the animation on the actor and resolves promises returned from `completed()`.
+	 */
+	public stop();
+
+	/**
+	 * Pauses the animation (sets animation speed to zero).
+	 */
+	public pause();
+
+	/**
+	 * Resumes the animation (sets animation speed to 1).
+	 */
+	public resume();
+
+	/**
+	 * Sets the animation time (units are in seconds).
+	 */
+	public setTime(time: number);
+
+	/**
+	 * Sets the playback speed.
+	 */
+	public setSpeed(speed: number);
+
+	/**
+	 * (Advanced) Sets the time, speed, and enabled state of an animation.
+	 */
+	public setState(state: SetAnimationStateOptions);
+
+	/**
+	 * @returns A Promise that resolves when the animation completes or is stopped.
+	 */
+	public completed(): Promise<void>;
+}
+```
+
+Then, on `Actor`:
+
+```ts
+/**
+ * getter that returns the array of animations defined on this actor.
+ */
+public get animations(): AnimationLike[];
+
+/**
+ * Get an animation by name.
+ */
+public animation(animationName: string): AnimationLike;
+```
+
+#### Examples
+
+```ts
+// Play the 'jump' animation, then do something when it's done.
+actor.animation("jump").start().then(() => console.log("done jumping"));
+
+// Play the 'victory' animation, and wait for it to complete.
+await actor.animation("victory").start();
+```
+
+### Interpolations
+
+#### Add `Context.interpolate` method
+
+Add an `interpolate` method to class `Context`. Calling `interpolate` initiates a lerp operation on the host and returns a Promise that will resolve after `seconds` time has elapsed or the interpolation is otherwise canceled.
+
+```ts
+/**
+ * Interpolates the field indicated by `selector` to `value` over `interval` seconds along `options.curve`.
+ * @param selector The selector of the field to interpolate. Get this value by calling actor.selector(fieldpath).
+ * @param value The value to interpolate to or by, depending on the value of options.basis. Can be a selector or a concrete value.
+ * @param interval The duration of the interolation in seconds. Pass in `null` to cancel an existing matching interpolation.
+ * @param options Optional parameters to further customize the interpolation.
+ * @param options.curve Specifies the interpolation's speed curve. This can be one of the predefined curve names, or a custom curve. Default: "linear".
+ * @param options.basis How to interpret `value`. If `options.basis` is "absolute" this function will interpolate to `value`. If `options.basis` is "relative", this function interpolates by `value` amount. Default: "absolute".
+ * @param options.iterationCount How many times to run the interpolation. Can be a number, or "infinite". Default: 1.
+ * @param options.direction The direction the interpolation should be played.
```

Comment:
Can't animate anywhere but forward without both a from and a to value.

Reply:
`from` and `to` are both required parameters, unless you're cancelling an existing interpolation (see examples section).

Reply:
Should be able to use the initial value of the field referenced by `selector` as the initial `from` value.